src/Databases/DatabaseReplicated.cpp

```diff
@@ -978,8 +978,51 @@ ASTPtr DatabaseReplicated::parseQueryFromMetadataInZooKeeper(const String & node
     return ast;
 }
 
+void DatabaseReplicated::dropReplica(
+    DatabaseReplicated * database, const String & database_zookeeper_path, const String & full_replica_name)
+{
+    assert(!database || database_zookeeper_path == database->zookeeper_path);
```

Comment:
So it will just fail in case of incorrect zkpath and debug build?

Reply:
It will not and there's a test for this

Reply:
Ah, yes, we pul `nullptr` as a pointer to a database. Sorry